source/material_model/rheology/frank_kamenetskii.cc

```diff
@@ -0,0 +1,118 @@
+/*
+  Copyright (C) 2019 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/frank_kamenetskii.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      FrankKamenetskii<dim>::FrankKamenetskii ()
+      {}
+
+
+
+      template <int dim>
+      double
+      FrankKamenetskii<dim>::compute_viscosity (const double reference_viscosity,
+                                                const double temperature,
+                                                const unsigned int composition) const
+      {
+        // Frank-Kamenetskii viscosity approximation
+        //    viscosity = ref_v * A * B * exp(E * (0.5-(T/ref_T)/2)))
+        // ref_v: reference viscosity,
+        // A: prefactor, B: pre-limiter, E: adjusted viscosity ratio,
+        // ref_T: reference temperature, T: temperature.
+
+        const double viscosity_frank_kamenetskii = reference_viscosity * prefactors_frank_kamenetskii[composition]
+                                                   * std::exp(viscosity_ratios_frank_kamenetskii[composition]*(0.5-temperature/reference_T/2));
```

Comment:
Would you be ok writing this as the following:
```suggestion
                                                   * std::exp(viscosity_ratios_frank_kamenetskii[composition] * 0.5 * (1.0 -temperature/reference_T));
```
This feels easier to read.